### PR TITLE
add TNS Descriptor support for HA JDBC Connections/PLUGIN-1095

### PR DESCRIPTION
**Pull request by jster1357**
add documentation and widget changes

### CODE REVIEW

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java [hunk 1]
```diff
@@ -57,14 +58,22 @@ public OracleAction(OracleActionConfig oracleActionConfig) {
...
       }
-      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT, host, port, database);
     }
 
 
+
+
+
+
+
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: minor nit: remove extra lines.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java [hunk 2]
```diff
@@ -57,14 +58,22 @@ public OracleAction(OracleActionConfig oracleActionConfig) {
 
     @Override
     public String getConnectionString() {
-      if (OracleConstants.SERVICE_CONNECTION_TYPE.equals(this.connectionType)) {
-        return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT,
-                             host, port, database);
+      if (OracleConstants.TNS_CONNECTION_TYPE.equals(this.connectionType)) {
+        return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);
+
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: minor nit: remove extra line.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java [hunk 3]
```diff
@@ -110,14 +110,21 @@ protected String getConnectionString(@Nullable String database) {
...
+              config.getHost(), config.getPort(), database);
     }
-    return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT,
-                         config.getHost(), config.getPort(), database);
   }
 
+
+
+
+
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java]: minor nit: remove extra lines.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java [hunk 4]
```diff
@@ -110,12 +110,15 @@ protected String getConnectionString(@Nullable String database) {
     if (database == null) {
       return config.getConnectionString();
     }
-    if (OracleConstants.SERVICE_CONNECTION_TYPE.equals(config)) {
+    if (OracleConstants.TNS_CONNECTION_TYPE.equals(config.getConnectionType())) {
       return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT, config.getHost(),
                            config.getPort(), database);
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java]: should this return
`return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);`

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java [hunk 5]
```diff
@@ -110,12 +110,15 @@ protected String getConnectionString(@Nullable String database) {
...
+    } else if (OracleConstants.SERVICE_CONNECTION_TYPE.equals(config.getConnectionType())) {
+      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java:114]: here it should be ` return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT, config.getHost(),
                           config.getPort(), database);`

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java [hunk 6]
```diff
@@ -56,14 +56,18 @@ public OracleConnectorConfig(String host, int port, String user, String password
...
+      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);
+    } else if (OracleConstants.SERVICE_CONNECTION_TYPE.equals(getConnectionType())) {
       return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT, host, getPort(), database);
+    } else {
+      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT, host, getPort(),database);
     }
-    return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SID_FORMAT, host, getPort(), database);
   }
 
+
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java]: minor nit: remove extra line.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java [hunk 7]
```diff
@@ -81,9 +85,7 @@ protected int getDefaultPort() {
     return 1521;
   }
 
-  public String getConnectionType() {
-    return connectionType;
-  }
+  public String getConnectionType() {return connectionType == null ? "service" : connectionType;}
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java]: connectionType should not be null. Did you observe it as `null` in any of your testing?

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java [hunk 8]
```diff
@@ -129,6 +135,7 @@ public void validate(FailureCollector collector) {
     public String getTransactionIsolationLevel() {
       return getConnection().getTransactionIsolationLevel();
     }
+
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleSource.java]: minor nit: remove extra line.

**Review by sagarkapare — COMMENTED** [review]
few minor comments.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java [hunk 9]
```diff
@@ -18,6 +18,7 @@
 
 import com.google.common.collect.ImmutableMap;
 import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
```
Comment (sagarkapare) [inline oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: unused import?

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java (hunk 4 above)
Comment (jster1357) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnector.java]: thanks for catching this.

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java (hunk 9 above)
Comment (jster1357) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleAction.java]: Yeah in earlier configs I was thinking about using it with a field. Not needed any more and will remove

File: oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java (hunk 7 above)
Comment (jster1357) [reply oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConnectorConfig.java]: I did not. It was always populated.

**Pull request closed without merging**